pkg/kubeapiserver/authenticator/config.go

```diff
@@ -148,33 +155,43 @@ func (config Config) New() (authenticator.Request, *spec.SecurityDefinitions, sp
 	// cache misses for all requests using the other. While the service account plugin
 	// simply returns an error, the OpenID Connect plugin may query the provider to
 	// update the keys, causing performance hits.
+	var updateAuthenticationConfig func(*apiserver.AuthenticationConfiguration) error
 	if config.AuthenticationConfig != nil {
-		for _, jwtAuthenticator := range config.AuthenticationConfig.JWT {
-			var oidcCAContent oidc.CAContentProvider
-			if len(jwtAuthenticator.Issuer.CertificateAuthority) > 0 {
-				var oidcCAError error
-				oidcCAContent, oidcCAError = dynamiccertificates.NewStaticCAContent("oidc-authenticator", []byte(jwtAuthenticator.Issuer.CertificateAuthority))
-				if oidcCAError != nil {
-					return nil, nil, nil, oidcCAError
-				}
-			}
-			oidcAuth, err := oidc.New(oidc.Options{
-				JWTAuthenticator:     jwtAuthenticator,
-				CAContentProvider:    oidcCAContent,
-				SupportedSigningAlgs: config.OIDCSigningAlgs,
-				DisallowedIssuers:    config.ServiceAccountIssuers,
-			})
+		var jwtAuthenticatorPtr atomic.Pointer[jwtAuthenticatorWithCancel]
+		var synchronousInitialization bool
+		updateAuthenticationConfig = func(authConfig *apiserver.AuthenticationConfiguration) error {
+			jwtAuthenticator, err := newJWTAuthenticator(ctx, authConfig, config.OIDCSigningAlgs, config.APIAudiences, config.ServiceAccountIssuers, synchronousInitialization)
 			if err != nil {
-				return nil, nil, nil, err
+				return err
+			}
+
+			oldJWTAuthenticator := jwtAuthenticatorPtr.Swap(jwtAuthenticator)
+			if oldJWTAuthenticator != nil {
+				// TODO maybe track requests so we know when this is safe to do
+				time.Sleep(JWTAuthenticatorSleepAfterSwapTime)
```

Comment:
this should do something like this, so we don't hang on server stop (I'm actually really surprised integration tests aren't complaining about this leaking goroutines):

```go
select {
case <-ctx.Done():
case <-time.After(time.Minute):
}
```